src/main/clojure/clara/rules/compiler.clj

```diff
@@ -1138,18 +1140,21 @@
             ;; will out perform equivalence checks.
             update-node->id (fn [m id]
                               (let [node (get id-to-condition-nodes id)
-                                    prev (get m node)]
+                                    ;; appending parents to allow for identical conditions that do not share the same
+                                    ;; parents, see Issue 433 for more information
+                                    node-with-parents (assoc node :parents (get backward-edges id))
```

Comment:
I'm trying to rationalize this.

If we had a case like:
```
a   b
| |
c
| |
d e
```
where `c` was a common condition with, `a` and `b` as parents and `d` and `e` as children, it isn't safe to reuse the _same_ `c` to propagate to the children `d` and `e` in all cases. Instead `c` needs to propagate to `d`, eg. _only_ when it is receiving propagation from it's parent `a`, and similarly, perhaps `e` should only be propagated to when we are going from `b` > `c` > `e`.

I think this makes sense. `c` can't be shared in the propagation - or even working memory structure, due to propagation semantics - when it is coming from 2 logically separate parent paths.

So far I'm onboard with this change.